src/app/components/map.tsx

```diff
@@ -0,0 +1,104 @@
+"use client";
+
+import "mapbox-gl/dist/mapbox-gl.css";
+import mapboxgl from "mapbox-gl";
+import { useEffect, useRef } from "react";
+import { NearbySightingsResponse } from "../recent/nearby/page";
+
+export default function MapComponent({
+  onDragEnd,
+  nearbySightings,
+}: {
+  nearbySightings: NearbySightingsResponse | undefined;
+  onDragEnd: (longitude: number, latitude: number) => void;
+}) {
+  const mapContainer = useRef<HTMLDivElement>(null);
+  const mapRef = useRef<mapboxgl.Map | null>(null);
+  const markersRef = useRef<mapboxgl.Marker[]>([]);
+
+  if (!process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN) {
+    throw new Error("Missing mapBox API token");
+  }
+
+  useEffect(() => {
+    const accessToken = process.env.NEXT_PUBLIC_MAPBOX_API_TOKEN;
+    mapboxgl.accessToken = `${accessToken}`;
+
+    if (mapContainer.current) {
+      const map = new mapboxgl.Map({
+        container: mapContainer.current,
+        style: "mapbox://styles/mapbox/outdoors-v12",
+        center: [-1.688074, 53.185335], // make center the most recent sighted bird or something
+        zoom: 10,
+        maxZoom: 15,
+        attributionControl: false,
+      });
+
+      mapRef.current = map;
+
+      map.addControl(new mapboxgl.NavigationControl(), "top-left");
+
+      let geolocate = new mapboxgl.GeolocateControl({
+        positionOptions: {
+          enableHighAccuracy: true,
+        },
+        trackUserLocation: true,
+      });
+
+      map.addControl(geolocate);
+
+      return () => map.remove();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (mapRef.current === null) return;
+
+    const handleMoveEvents = () => {
+      if (mapRef.current) {
+        const longitude = mapRef.current.getCenter().lng;
+        const latitude = mapRef.current.getCenter().lat;
+        onDragEnd(longitude, latitude);
+      }
+    };
+
+    mapRef.current.on("dragend", () => {
+      handleMoveEvents();
+    });
+
+    return () => {
+      if (mapRef.current) {
+        mapRef.current.off("dragend", handleMoveEvents);
+      }
+    };
+  }, [onDragEnd]);
```

Comment:
The code that initialises the map is now separated from the getCenter event listeners, the dependancy is now here where it's needed. 

Resulting in: The Map does not refresh every time a dragend event occurs and the dependencies are where they need to be..